api/src/main/java/org/hawkular/inventory/api/filters/Filter.java

```diff
@@ -42,9 +42,10 @@
 public abstract class Filter {
     private static final Filter[] EMPTY = new Filter[0];
 
-    Filter() {
-
-    }
+//    we should not prohibit creating sub-classes of Filter in the implementations of this API
```

Comment:
I guess Javadoc is the best solution here ;)

The in-memory application is basically the same idea as `GremlinPipeline.filter()` method. It just applies a predicate to the intermediate results of the pipeline.